src/app/check-in/check-in.component.html

```diff
@@ -25,28 +25,31 @@
 <div class="child-container" (keydown)="keyDownFunction($event)">
 
   <div class="row">
-    <!-- <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
-      <label><input id="usercode" type="text" autofocus [(ngModel)]='patientId' placeholder="Usercode" > </label>
-    </div>-->
-  </div>
-    <div class="row">
-      <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
-          <label><input id="firstname" type="text" autofocus [(ngModel)]='firstName' placeholder="First Name" > </label>
-      </div>
+
+    <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
+      <label><input id="usercode" type="text" autofocus [(ngModel)]='firstName' placeholder="First Name" > </label>
     </div>
-  <br>
-  <div class="row">
+
+    <br>
+
     <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
-      <label><input id="lastname" type="text" autofocus [(ngModel)]='lastName' placeholder="Last Name" > </label>
+      <label><input id="usercode" type="text" autofocus [(ngModel)]='lastName' placeholder="Last Name" > </label>
     </div>
-  </div>
 
-  <br>
-  <div class="row">
+    <br>
+
+    <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
+      <label><input id="usercode" type="text" autofocus [(ngModel)]='email' placeholder="Email" > </label>
+    </div>
+
+    <br>
+
     <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
-      <label><input id="lastfourssn" type="text" autofocus [(ngModel)]='lastFourSS' placeholder="Last 4 digits of SSN" > </label>
+      <label><input id="usercode" type="text" autofocus [(ngModel)]='patientId' placeholder="Usercode" > </label>
     </div>
+
   </div>
+
     <div class="row">
       <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
         <div *ngIf="!authenticationFlag" class="failLogIn">One or more of the above fields is incomplete.</div>
```

Comment:
Point Error message to a component variable held here to be an error message to tell the user what rule they are violating:
```suggestion
        <div *ngIf="!authenticationFlag" class="failLogIn">{{errorMessage}}</div>
```

Reply:
I agree with Elijah's suggestion of holding the error variable.